pontoon/sync/vcs/models.py

```diff
@@ -431,6 +438,69 @@ def resources_for_path(self, path):
                     yield os.path.join(root, filename)
 
 
+class VCSConfiguration(object):
+    """
+    Container for the project configuration, provided by the optional
+    configuration file.
+
+    For more information, see:
+    https://moz-l10n-config.readthedocs.io/en/latest/fileformat.html.
+    """
+
+    def __init__(self, vcs_project):
+        self.vcs_project = vcs_project
+        self.configuration_file = vcs_project.db_project.configuration_file
+
+    @cached_property
+    def configs(self):
+        """Return parsed project configuration file."""
+        path = os.path.join(
+            self.vcs_project.db_project.source_repository.checkout_path,
+            self.configuration_file
+        )
+        parser = TOMLParser()
+        return parser.parse(path).configs
+
+    @cached_property
+    def paths(self):
+        """Return paths specified in the configurations."""
+        return [
+            paths
+            for configuration in self.configs
+            for paths in configuration.paths
+        ]
```

Comment:
The configurations can also be locale-dependent. For example, we could have a smaller list of locales in `_configs/devtools-client.toml`.